### PR TITLE
Feature/chat

### PR DESCRIPTION
**Pull request by olee**
Finished implementation, ready for merging.

### CODE REVIEW

File: src/main/java/com/forgeessentials/chat/ModuleChat.java [hunk 1]
```diff
@@ -46,202 +66,446 @@
...
+        setupChatReplacements();
     }
 
-    @SubscribeEvent
-    public void postLoad(FEModulePostInitEvent e)
+    public void setupChatReplacements()
     {
-        mailsystem = new MailSystem();
-
+        chatConstReplacements.put("smile", "\u263A");
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/chat/ModuleChat.java]: Can this be moved to another class, maybe as an enum?

File: src/main/java/com/forgeessentials/chat/irc/IrcHandler.java [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    public boolean isConnected()
+    {
+        return bot != null && bot.isConnected();
+    }
+
+    /* ------------------------------------------------------------ */
+
+    @Override
+    public void load(Configuration config, boolean isReload)
+    {
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/chat/irc/IrcHandler.java]: Bot commands should be configurable. Some servers may want their IRC bot to only relay messages,

File: src/main/java/com/forgeessentials/permissions/core/ZonedPermissionHelper.java [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package com.forgeessentials.permissions.core;
```
Comment (luacs1998) [inline src/main/java/com/forgeessentials/permissions/core/ZonedPermissionHelper.java]: Commit this either directly to develop or make a separate PR.

**Comment by luacs1998** [comment]
Does this fix #1342?

And, is it possible to implement #1429 and #1445 as part of this PR?

File: src/main/java/com/forgeessentials/chat/ModuleChat.java (hunk 1 above)
Comment (olee) [inline src/main/java/com/forgeessentials/chat/ModuleChat.java]: I moved it from FunctionHelper over here.
If you want to change it, do it yourself once the PR is merged.
I don't have time for these small and pretty useless changes.

File: src/main/java/com/forgeessentials/chat/irc/IrcHandler.java (hunk 2 above)
Comment (olee) [inline src/main/java/com/forgeessentials/chat/irc/IrcHandler.java]: Yeah I am going to add a few more config options to the bot.

**Comment by olee** [comment]
This fixes #1342

File: src/main/java/com/forgeessentials/chat/ModuleChat.java (hunk 1 above)
Comment (luacs1998) [inline src/main/java/com/forgeessentials/chat/ModuleChat.java]: OK.

**Comment by olee** [comment]
@luacs1998 take a look at it if you want and when your done, merge.

File: src/main/java/com/forgeessentials/chat/command/CommandTimedMessages.java [hunk 4]
```diff
@@ -130,7 +124,7 @@ public String getPermissionNode()
     @Override
     public String getCommandUsage(ICommandSender sender)
     {
-        return "/automessage [select|broadcast|add|del] Select, broadcast, add or remove messages";
+        return "/TimedMessageHandler [select|broadcast|add|del] Select, broadcast, add or remove messages";
```
Comment (peddermaster2) [inline src/main/java/com/forgeessentials/chat/command/CommandTimedMessages.java]: isn't this supposed to be `/timedmessage ...` as in [Line 23](https://github.com/ForgeEssentials/ForgeEssentialsMain/pull/1532/files#diff-7fd9849db0f1ec7af179f816c44004a9R23)
